messaging/src/main/java/org/axonframework/common/caching/Cache.java

```diff
@@ -59,6 +60,17 @@ public interface Cache {
      */
     boolean putIfAbsent(Object key, Object value);
 
+    /**
+     * Returns the value under the given {@code key} in the cache. If there is no value present, will invoke the given
+     * {@code valueSupplier}, put the value in the cache and return the produced value.
+     *
+     * @param key           The key under which to store the item
+     * @param valueSupplier A supplier that lazily supplies the value when necessary
+     * @return The value that is in the cache after the operation. This can be the original value or the one supplied by
+     * the {@code valueSupplier}.
+     */
+    <T> T computeIfAbsent(Object key, Supplier<T> valueSupplier);
```

Comment:
> Also, it seems some implementations already follow the same pattern, which can be replaced by the default implementation.

Which components/methods/implementations are you exactly referring to in this case, @abuijze?

Reply:
The JCacheAdapter and EHCacheAdapter both follow the same pattern. Instead of implementing the methods there, they could use a default implementation on the interface.

Reply:
Gotcha, thanks for specifying that.